File: flutter_app/lib/models/order.dart
class Order {
  final String orderId;
  final String status;
  final double total;
  final String createdAt;

  Order({
    required this.orderId,
    required this.status,
    required this.total,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      orderId: json['order_id'] ?? '',
      status: json['status'] ?? '',
      total: (json['total'] ?? 0).toDouble(),
      createdAt: json['created_at'] ?? '',
    );
  }
}

File: go_backend/models/order.go
package models

import "time"

type Order struct {
	ID                  string    `json:"id"`
	UserID              string    `json:"user_id"`
	Status              string    `json:"status"`
	TotalAmount         float64   `json:"total_amount"`
	PaypalTransactionID string    `json:"paypal_transaction_id"`
	CreatedAt           time.Time `json:"created_at"`
	UpdatedAt           time.Time `json:"updated_at"`
}

type OrderItem struct {
	ID              string  `json:"id"`
	OrderID         string  `json:"order_id"`
	ProductID       string  `json:"product_id"`
	Quantity        int     `json:"quantity"`
	PriceAtPurchase float64 `json:"price_at_purchase"`
}
